src/sentry/utils/sdk_crashes/sdk_crash_detection.py

```diff
@@ -44,29 +50,39 @@ def detect_sdk_crash(self, event: Event) -> None:
 
         context = get_path(event.data, "contexts", "sdk_crash_detection")
         if context is not None and context.get("detected", False):
-            return
+            return None
 
         is_unhandled = (
             get_path(event.data, "exception", "values", -1, "mechanism", "data", "handled") is False
         )
         if is_unhandled is False:
-            return
+            return None
 
         frames = get_path(event.data, "exception", "values", -1, "stacktrace", "frames")
         if not frames:
-            return
+            return None
 
         if self.cocoa_sdk_crash_detector.is_sdk_crash(frames):
-            sdk_crash_event = self.event_stripper.strip_event_data(event)
+            sdk_crash_event_data = self.event_stripper.strip_event_data(event)
 
             set_path(
-                sdk_crash_event.data, "contexts", "sdk_crash_detection", value={"detected": True}
+                sdk_crash_event_data, "contexts", "sdk_crash_detection", value={"detected": True}
+            )
+
+            if settings.SDK_CRASH_DETECTION_PROJECT_ID is None:
+                return None
```

Comment:
If `SDK_CRASH_DETECTION_PROJECT_ID` is `None`, there is no benefit in running any of the SDK crash reporting code, or is there? If not, I would move this check to the top-level function in `post_process`.